### ISSUE TITLE
Test case is running fine in Electron but not able to run in Chrome (using Cypress run it is working fine ,but with using CLI It is failing)

### ISSUE THREAD

**Issue by Gaurav1115-dev** [cypress-io/cypress#21183]
### Current behavior

  Running:  delta_demo/homepage.js                                                          (1 of 1)
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating

  Home Page Validation
    1) Page Navigation 

  0 passing (3s)
  1 failing

  1) Home Page Validation
       Page Navigation :
     CypressError: `cy.visit()` failed trying to load:

https://www.delta.com/

We attempted to make an http request to this URL but the request failed without a response.

We received this error at the network level:

  > Error: Exceeded maxRedirects. Probably stuck in a redirect loop https://www.delta.com/content/www/en_US/system-unavailable1.html

Common situations why this would fail:
  - you don't have internet access
  - you forgot to run / boot your web server
  - your web server isn't accessible
  - you have weird network configuration settings on your computer
      at http://localhost:49900/__cypress/runner/cypress_runner.js:159325:82
      at visitFailedByErr (http://localhost:49900/__cypress/runner/cypress_runner.js:158675:12)
      at http://localhost:49900/__cypress/runner/cypress_runner.js:159324:11
      at tryCatcher (http://localhost:49900/__cypress/runner/cypress_runner.js:13022:23)
      at Promise._settlePromiseFromHandler (http://localhost:49900/__cypress/runner/cypress_runner.js:10957:31)
      at Promise._settlePromise (http://localhost:49900/__cypress/runner/cypress_runner.js:11014:18)
      at Promise._settlePromise0 (http://localhost:49900/__cypress/runner/cypress_runner.js:11059:10)
      at Promise._settlePromises (http://localhost:49900/__cypress/runner/cypress_runner.js:11135:18)
      at _drainQueueStep (http://localhost:49900/__cypress/runner/cypress_runner.js:7729:12)
      at _drainQueue (http://localhost:49900/__cypress/runner/cypress_runner.js:7722:9) 
      at Async.../../node_modules/bluebird/js/release/async.js.Async._drainQueues (http://localhost:49900/__cypress/runner/cypress_runner.js:7738:5)
      at Async.drainQueues (http://localhost:49900/__cypress/runner/cypress_runner.js:7608:14)
  From Your Spec Code:
      at Context.eval (http://localhost:49900/__cypress/tests?p=cypress\integration\delta_demo\homepage.js:127:8)

  From Node.js Internals:
    Error: Exceeded maxRedirects. Probably stuck in a redirect loop https://www.delta.com/content/www/en_US/system-unavailable1.html
        at processRedirect (C:\Users\ergau\AppData\Local\Cypress\Cache\9.5.4\Cypress\resources\app\node_modules\@cypress\request\lib\redirect.js:104:23)
        at tryCatcher (C:\Users\ergau\AppData\Local\Cypress\Cache\9.5.4\Cypress\resources\app\packages\server\node_modules\bluebird\js\release\util.js:16:23)
        at Promise._settlePromiseFromHandler (C:\Users\ergau\AppData\Local\Cypress\Cache\9.5.4\Cypress\resources\app\packages\server\node_modules\bluebird\js\release\promise.js:547:31)
        at Promise._settlePromise (C:\Users\ergau\AppData\Local\Cypress\Cache\9.5.4\Cypress\resources\app\packages\server\node_modules\bluebird\js\release\promise.js:604:18)   
        at Promise._settlePromise0 (C:\Users\ergau\AppData\Local\Cypress\Cache\9.5.4\Cypress\resources\app\packages\server\node_modules\bluebird\js\release\promise.js:649:10)  
        at Promise._settlePromises (C:\Users\ergau\AppData\Local\Cypress\Cache\9.5.4\Cypress\resources\app\packages\server\node_modules\bluebird\js\release\promise.js:729:18)  
        at _drainQueueStep (C:\Users\ergau\AppData\Local\Cypress\Cache\9.5.4\Cypress\resources\app\packages\server\node_modules\bluebird\js\release\async.js:93:12)
        at _drainQueue (C:\Users\ergau\AppData\Local\Cypress\Cache\9.5.4\Cypress\resources\app\packages\server\node_modules\bluebird\js\release\async.js:86:9)
        at Async._drainQueues (C:\Users\ergau\AppData\Local\Cypress\Cache\9.5.4\Cypress\resources\app\packages\server\node_modules\bluebird\js\release\async.js:102:5)
        at Immediate.Async.drainQueues [as _onImmediate] (C:\Users\ergau\AppData\Local\Cypress\Cache\9.5.4\Cypress\resources\app\packages\server\node_modules\bluebird\js\release\async.js:15:14)

### Test code to reproduce

Use the below Repo for re producing the issue(Use Navigation.js and Homepage .js
https://github.com/Gaurav1115-dev/cypressDelta

### Cypress Version

9.4.0

**Comment by Gaurav1115-dev** [comment]
Hi Cypress-Io/Cypress,

You can use "delta-vacations" instead of "Travel". and try to run the test
case using CLI (with headless mode ). I'm still facing the same problem.

Thanks
Gaurav

**Comment by miracllife** [comment]
i think it is similar to this one https://github.com/cypress-io/cypress/issues/6202 , Also i have come across this problem.

**Comment by miracllife** [comment]
> i think it is similar to this one #6202 , Also i have come across this problem.

but this issue is not resolve but closed

**Comment by nabeelDanishC** [comment]
@Gaurav1115-dev @davidmunechika 

Did you find a solution to this problem? My problem is very similar to yours. 

Running a smoke test using Cypress UI on Chrome works perfectly fine, but when I run Cypress headlessly on Chrome, the test fails on a `url()` check. Furthermore, Running the same test on different browsers like Electron or Edge (Headlessly) seems to work as well.

**Comment by abrilmaestre** [comment]
I have the same problem. With chrome/firefox in headless mode test fails but with electron it works.

**Comment by nagash77** [comment]
We are still waiting on a reproducible example to investigate this further.  @abrilmaestre I would suggest you open a new issue with your specific use case and example.

**Comment by nagash77** [comment]
Right now there doesn't seem to be enough information to reproduce the problem on our end. We'll have to close this issue until we can reproduce it. This does not mean that your issue is not happening - it just means that we do not have a path to move forward.

Please open a new issue with a reproducible example and link to this issue. Here are some tips for providing a [Short, Self Contained, Correct, Example](http://sscce.org/) and our own [Troubleshooting Cypress](https://on.cypress.io/troubleshooting) guide.
